src/cupbearer/data/backdoors.py

```diff
@@ -150,95 +144,75 @@ def warping_field(self) -> np.ndarray:
 
     def init_warping_field(self, px: int, py: int):
         logger.debug("Generating new warping field")
-        field = np.stack(
-            [
-                map_coordinates(  # map_coordinates and upsample diffs slightly
-                    input=grid,
-                    coordinates=np.mgrid[
-                        0 : (self.control_grid_width - 1) : (py * 1j),
-                        0 : (self.control_grid_width - 1) : (px * 1j),
-                    ],
-                    order=3,
-                    mode="nearest",
-                )
-                for grid in self.control_grid
-            ],
-            axis=0,
-        )
+        control_grid = self.control_grid
+        assert control_grid.ndim == 3
+        # upsample expects a batch dimesion, so we add a singleton. We permute after
+        # upsampling, since grid_sample expects the length-2 axis to be the last one.
+        field = F.upsample(
```

Comment:
```suggestion
        field = F.interpolate(
```
upsample has become deprecated, but should just be a straight replacement with interpolate